pkg/ttl/cache/table.go

```diff
@@ -185,28 +183,40 @@ func (t *PhysicalTable) ValidateKeyPrefix(key []types.Datum) error {
 	return nil
 }
 
-// EvalExpireTime returns the expired time
+// EvalExpireTime returns the expired time.
+// It uses the global timezone to compute the expired time.
+// Then we'll reset the returned expired time to the same timezone as the input `now`.
 func (t *PhysicalTable) EvalExpireTime(ctx context.Context, se session.Session,
 	now time.Time) (expire time.Time, err error) {
-	tz := se.GetSessionVars().Location()
-
 	expireExpr := t.TTLInfo.IntervalExprStr
 	unit := ast.TimeUnitType(t.TTLInfo.IntervalTimeUnit)
 
-	var rows []chunk.Row
-	rows, err = se.ExecuteSQL(
-		ctx,
-		// FROM_UNIXTIME does not support negative value, so we use `FROM_UNIXTIME(0) + INTERVAL <current_ts>`
-		// to present current time
-		fmt.Sprintf("SELECT FROM_UNIXTIME(0) + INTERVAL %d SECOND - INTERVAL %s %s", now.Unix(), expireExpr, unit.String()),
-	)
-
+	y, m, d, nanosecond, _, err := types.ParseDurationValue(unit.String(), expireExpr)
 	if err != nil {
-		return
+		return time.Time{}, err
 	}
 
-	tm := rows[0].GetTime(0)
-	return tm.CoreTime().GoTime(tz)
+	// Use the global time zone to compute expire time.
+	// Different timezones may have different results event with the same "now" time and TTL expression.
+	// Consider a TTL setting with the expiration `INTERVAL 1 MONTH`.
+	// If the current timezone is `Asia/Shanghai` and now is `2021-03-01 00:00:00 +0800`
+	// the expired time should be `2022-02-01 00:00:00 +0800`, corresponding to UTC time `2022-01-31 16:00:00 UTC`.
+	// But if we use the `UTC` time zone, the current time is `2021-02-28 16:00:00 UTC`,
+	// and the expired time should be `2021-01-28 16:00:00 UTC` that is not the same the previous one.
```

Comment:
Good catch and well written comment!
Should we mention this in the manual as well, that the TTL will be executed in the global time zone?
I think the other way around with a global time zone west of GMT is more confusing, where '2021-02-28 16:00:00 -08:00' will delete up to '2021-02-01 00:00:00 UTC', if the global time zone is UTC?
Meaning that in the users local time zone, it will not have access to one month old data, i.e. the user is still in Februrary, but cannot access the last data in January.
Probably should be mentioned for any unit that includes 'Month' and 'Quarter'?

Reply:
Thanks for your comment. It's better to mention it in the manual.